shotover/src/transforms/kafka/sink_cluster/scram_over_mtls.rs

```diff
@@ -12,11 +13,92 @@ use kafka_protocol::{
     messages::{ApiKey, CreateDelegationTokenRequest, RequestHeader},
     protocol::{Builder, StrBytes},
 };
-use rand::prelude::SliceRandom;
 use rand::rngs::SmallRng;
+use rand::{prelude::SliceRandom, SeedableRng};
 use serde::{Deserialize, Serialize};
-use std::{sync::Arc, time::Duration};
+use std::{collections::HashMap, sync::Arc, time::Duration};
 use tokio::sync::Notify;
+use tokio::sync::{mpsc, oneshot};
+
+pub struct TokenRequest {
+    username: String,
+    response_tx: oneshot::Sender<DelegationToken>,
+}
+
+#[derive(Clone)]
+pub struct TokenTask {
+    tx: mpsc::Sender<TokenRequest>,
+}
+
+impl TokenTask {
+    #[allow(clippy::new_without_default)]
+    pub fn new(
+        mtls_connection_factory: ConnectionFactory,
+        mtls_port_contact_points: Vec<String>,
+    ) -> TokenTask {
+        let (tx, mut rx) = mpsc::channel::<TokenRequest>(100);
```

Comment:
Why 100?

Reply:
I just picked the value at random.
An alternative would be to make it unlimited but I feel having some kind of limit in place would be better off.
This code is only called on connection creation so the only scenario I can see it coming into place is if a client startsup and creates a whole bunch of connections at once.

I think I'll bump the value to 1000, since processing only 100 incoming connections at once does feel possibly a bit low.